src/main/java/com/criticalsoftware/Product.java

```diff
@@ -1,67 +1,77 @@
 package com.criticalsoftware;
+
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.Pattern;
+import jakarta.validation.constraints.Size;
 import org.bson.types.ObjectId;
-import java.awt.image.BufferedImage;
 
 public class Product {
-
     private ObjectId id;
+
+    @NotBlank(message = "Product name is mandatory")
+    @Size(max = 30, message = "Product name must be less than or equal to 30 characters")
```

Comment:
These validations should be in the request, not here.

Reply:
Done!